File: src/template/java/com/my/WebPageTemplate.java
package src.template.java.com.my;

import src.chainOfResponsibility.java.com.my.ChainLink;
import src.chainOfResponsibility.java.com.my.RoleCheckLink;
import src.chainOfResponsibility.java.com.my.TokenCheckLink;
import src.chainOfResponsibility.java.com.my.UserExistsCheckLink;

public abstract class WebPageTemplate {
    public void show() {
        check();
        header();
        content();
        footer();
    }

    public void check() {
        if (ChainLink.initialize(
                new UserExistsCheckLink(),
                new TokenCheckLink(),
                new RoleCheckLink()
        ).check("1", "1", "token")) {
            System.out.println("logged in");
        } else {
            System.out.println("not logged in");
        }
    }

    public abstract void header();
    public abstract void content();
    public abstract void footer();
}

File: src/template/python/WebPageTemplate.py
from abc import ABC, abstractmethod

from src.chainOfResponsibility.python.ChainLink import ChainLink
from src.chainOfResponsibility.python.RoleCheckLink import RoleCheckLink
from src.chainOfResponsibility.python.TokenCheckLink import TokenCheckLink
from src.chainOfResponsibility.python.UserExistsCheckLink import UserExistsCheckLink


class WebPageTemplate(ABC):
    def show(self):
        self.check()
        self.header()
        self.content()
        self.footer()

    def check(self):
        if (ChainLink.initialize(
            UserExistsCheckLink(),
            TokenCheckLink(),
            RoleCheckLink()
        ).check("1", "1", "token")):
            print("logged in")
        else:
            print("not logged in")

    @abstractmethod
    def header(self): pass
    @abstractmethod
    def content(self): pass

    @abstractmethod
    def footer(self): pass
